src/model/encoding/convertFromHTMLToContentBlocks.js

```diff
@@ -517,17 +585,48 @@ function convertFromHTMLtoContentBlocks(
           return CharacterMetadata.create(data);
         })
       );
+      var key  = nullthrows(chunk).keys[ii];
       start = end + 1;
 
+      var blockType = nullthrows(chunk).blocks[ii].type;
+      var blockConfig = blockRenderMap.get(blockType);
+      var nextChunkKey = nullthrows(chunk).keys[ii+1];
+      var hasChildren = key && nextChunkKey && nextChunkKey.indexOf(key + '/') !== -1;
+
+      if (blockConfig && blockConfig.nestingEnabled && hasChildren) {
+        var character = '';
+        var blockKey = key || generateRandomKey();
+
+        if ((hasChildren && textBlock) || !hasChildren) {
+          return [
+            new ContentBlock({
+              key: blockKey,
+              type: blockType,
+              depth: nullthrows(chunk).blocks[ii].depth,
+              text: character,
+              characterList: List(Repeat(CharacterMetadata.create(), character.length))
+            }),
+            new ContentBlock({
+              key: blockKey + '/' + generateRandomKey(),
+              type: 'unstyled',
+              text: textBlock,
+              characterList,
+            })
+          ];
+        }
+      }
+
       return new ContentBlock({
-        key: generateRandomKey(),
-        type: nullthrows(chunk).blocks[ii].type,
+        key: key || generateRandomKey(),
```

Comment:
will we ever get to the stage where we would need to generate a new random key ? we should write some documentation on the side explaining under what circumstances this would happen